flask/cli.py

```diff
@@ -874,6 +874,50 @@ def routes_command(sort, all_methods):
 ))
 
 
+def run_from_app_run(app, host=None, port=None, debug=None):
+    if 'FLASK_APP' not in os.environ:
+        # attempt to determine the value of FLASK_APP through introspection
+        import inspect
+
+        app_name = None
+        for name, value in inspect.getmembers(sys.modules['__main__']):
+            if value == app:
+                app_name = name
+                break
+        if app_name is None:
+            return False
+
+        os.environ['FLASK_APP'] = '{}:{}'.format(sys.argv[0], app_name)
+
+    if 'FLASK_ENV' not in os.environ:
+        os.environ['FLASK_ENV'] = 'development' if debug else 'production'
+
+    # look for the flask executable in the same location where python is
+    executable = os.path.join(os.path.dirname(sys.executable), 'flask')
```

Comment:
I think this will fail on Windows, where it's "flask.exe".

Reply:
I'll find a Windows machine to test this, but I don't think it will fail. I don't think you have to provide the extension, the OS tries a bunch of known executable extensions such as .exe, .com and .bat until one matches.

Reply:
@davidism no problem at all running this patch on Windows.